01-ruby-fundamentals/07-oo-classes.md

```diff
@@ -115,33 +225,64 @@ class Chair
 end
 ```
 
-### Use `attr` Methods to Avoid Repetition!
-The code above allows us to read/get and write/set the color and name properties in the `Chair` _class_. This is done so frequently that Ruby added some syntactic sugar to help us out. Enter `attr_reader` and `attr_writer`:
+Now when we create a new `Chair`, we can give it some initial settings.
+
+```ruby
+kari_chair = Chair.new('swivel', 'black')
+puts kari_chair.style     # => swivel
+```
+
+Much more concise! This pattern of passing in a bunch of values for instance variables is a very common one, but the constructor is just another method, so you can write any code you need in it. You can even call other methods!
+
+Another advantage of setting instance variables in the constructor is that we know those variables will always have a value. By making it impossible to have a `Chair` without an associated `style` and `color`, we can save ourselves all sorts of frustration later on.
+
+### Use Macros to Avoid Repetition
+
+The code above allows us to read/get and write/set the color and style properties in the `Chair` _class_. This is done so frequently that Ruby added some syntactic sugar to help us out. Enter two _macros_, `attr_reader` and `attr_writer`:
 
 ```ruby
 class Chair
-  attr_reader :name, :color
-  attr_writer :name, :color
+  attr_reader :style, :color
```

Comment:
Could we show it with only one attribute for at least one so they don't get the impression that you have to have them for every instance variable?